### PR TITLE
Use explicit 0 instead of AsyncoverflowPolicy::block_retry to avoid gcc 5.4 link error

### PR DESCRIPTION
**Pull request by bodgergely**
Fix gcc 5.4 link error

### CODE REVIEW

File: src/pyspdlog.cpp [hunk 1]
```diff
@@ -403,7 +403,7 @@ PYBIND11_MODULE(spdlog, m) {
 
     m.def("set_async_mode", set_async_mode, 
             py::arg("queue_size") = 1 << 16,
-            py::arg("async_overflow_policy") = AsyncOverflowPolicy::block_retry,
+            py::arg("async_overflow_policy") = 0,    // 0 equals AsyncOverflowPolicy::block_retry - using block_retry results in gcc 5.4 compiler error
```
Comment (rafalmiel) [inline src/pyspdlog.cpp]: is this backward compatible?

Comment (rafalmiel) [reply src/pyspdlog.cpp]: Oh I see its equivalent

Comment (bodgergely) [reply src/pyspdlog.cpp]: pyspdlog is POSIX compliant

Comment (rafalmiel) [reply src/pyspdlog.cpp]: We should add jenkins test in the future

**Pull request merged**